### ISSUE TITLE
--max-peers not honored if --min-peers is not specified

### ISSUE THREAD

**Issue by GoodMirek** [paritytech/parity#7576]
> I'm running:
>
> - **Which Parity version?**:
```
parity --version
Parity
  version Parity/v1.10.0-unstable-ad14e656f-20180115/x86_64-linux-gnu/rustc1.22.1
```
> - **Which operating system?**: Linux
```
uname -a
Linux hostx 4.14.13-300.fc27.x86_64 #1 SMP Thu Jan 11 04:00:01 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
> - **How installed?**: from source
> - **Are you fully synchronized?**: no
> - **Which network are you connected to?**: ethereum
> - **Did you try to restart the node?**: yes

Running command:
```
parity --max-peers 10
```
does not limit the maximum number of peers, unless `--min-peers` is specified too.
It seems that the default value of min-peers prevails CLI option of `--max-peers`. When `--max-peers` is specified to be lower than the min-peers default value, then it is not honored.

Expected behavior is that explicitly expressed options always prevails defaults, so `--max-peers` should have been honored.

File `~/.local/share/io.parity.ethereum/config.toml` does not exist on the affected system.

**Comment by 5chdn** [comment]
That's intended, you can not have less than minimum peers :)

**Comment by GoodMirek** [comment]
I would think that explicitly specified options should prevail the defaults.

In this case, I would expect that specifying `--max-peers` without providing `--min-peers` should automatically lower min-peers to be the same or lower than the max-peers. In order not to constantly connect/disconnect peers, which behavior is probably costly (is it?), I can imagine that it can work in e.g. following way:
If `--min-peers` is omitted and only `--max-peers` is specified then min-peers could be set to a half of max-peers, but not less than e.g. 10 and never more than max-peers.

**Comment by niklasad1** [comment]
Solved in #8087
